pkg/tools/dnscontext/manager.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dnscontext
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+)
+
+// Manager provides API for storing/deleting dnsConfigs. Can represent the configs in caddyfile format.
+// Can be used from different goroutines
+type Manager interface {
```

Comment:
Is there a reason we are pulling this out into a separate thing?

Reply:
> Is there a reason we are pulling this out into a separate thing?

Yes, there I extract DNS tools to make the structure of `connectioncontext/dnscontext` pkg more clear. For example, pkg `connectioncontext/dnscontext` has only server, client and test files, and all depend on tools located in pkg `tools/...`.